src/game/character/skeleton.gd

```diff
@@ -0,0 +1,56 @@
+extends Node2D
+
+# --Private Variables--
+
+# Set the accepted file extensions to ".png"
+var extensions: Array = [".png"]
+
+# Directory of color maps. File names within should match shader names.
+var colorMapDir: Dictionary = {
+	"Color Maps": "res://game/character/assets/colormaps"
+}
+
+# Set the list of colormaps from the color map directory
+onready var colorShaders = Resources.list(colorMapDir, extensions)
+
+# Dictionary mapping each asset to a node path
+var nodeStructure: Dictionary = {
+	"Body": "Body",
+	"Clothes": "Clothes",
+	"Face Wear": "FaceWear",
+	"Facial Hair": "FacialHair",
+	"Hat/Hair": "HatHair",
+	"Mouth": "Mouth",
+	"Left Arm": "LeftArm",
+	"Left Leg": "LeftLeg",
+	"Right Arm": "RightArm",
+	"Right Leg": "RightLeg",
+	"Pants": "Pants",
+}
+
+# --Public Functions--
+
+# Apply config from appearance's variables
+func applyAppearance(outfit: Dictionary, colors: Dictionary) -> void:
```

Comment:
Put an assert either here (assert len(outfit)>0 or something, I don't know how "isempty" work in gdscript), so in the future we'll find it if something is doing a f*cky-wacky.